packaging/version.py

```diff
@@ -198,7 +198,12 @@ class Version(_BaseVersion):
 
     def __init__(self, version):
         # Validate the version and parse it into pieces
-        match = self._regex.search(version)
+        try:
+            match = self._regex.search(version)
+        except TypeError:
+            raise InvalidVersion(
+                "Invalid type: '{0}' is not a string".format(type(version))
```

Comment:
@dstufft I went ahead and changed to catch `TypeError`, but something bugs me a bit:
`InvalidVersion` inherits from `ValueError`.
So raising something of the `ValueError` familly on a `TypeError` feels weird.
I could make `InvalidVersion` inherit from both `ValueError` and `TypeError`,
or make a seperate exception and add it's usage all around.
Or, if you feel this is ok and I'm being too nitpicky with my things, that's ok.